### DISCUSSION TITLE
Integration Tests in TCA

### DISCUSSION THREAD

**Discussion by andrewjroberts** [pointfreeco/swift-composable-architecture#2005] [Q&A]
Hi TCA friends,

Does anyone have any general recommendations for implementing integration tests in a TCA app? My goal is to test my app's live FirebaseAuth signup flow. Thus far, haven't really found an example of TCA test code that attempts this sort of live testing. 

And just to be clear to any readers, I'm using the term "integration tests" in the way that was referenced in a previous discussion here: https://github.com/pointfreeco/swift-composable-architecture/discussions/1975#discussioncomment-5308931
> "testing code that interacts with outside systems, e.g. making live network requests, running code in a host (simulator, browser, etc.), and so on."

**Comment by mbrandonw** [comment]
Hi @andwrobs, have you tried writing an integration test and run into any problems so far? That might help guide the discussion.

It is of course possible to use live dependencies in a TCA unit test. You can just override the dependencies on the `TestStore` to override any of them with their `.liveValue`:

```swift
let store = TestStore(…) {
  $0.authClient = .liveValue
}
```

However, from what I understand of Firebase there is a decent amount of bootstrapping necessary to get it going. IIRC you need an info.plist prepared and then to call some configuration methods at the entry point of the application. I'm guessing you will need to repeat all of that in your test too.

**Reply by andrewjroberts** [reply]
Hi @mbrandonw, this helps a ton. I wasn't sure if using `TestStore` + TCA unit tests were appropriate for testing live dependencies, if there were libraries other than XCTest that might make things easier, etc... so it's good to get the validation on the approach here.

> just override the dependencies on the TestStore to override any of them with their .liveValue

This is a perfect place for me to start. I'll followup on this thread and will hopefully have some progress to share to help others w.r.t. the Firebase bootstrapping process.

Thanks for the response!
